hazelcast/src/test/java/com/hazelcast/client/protocol/compatibility/ReferenceObjects.java

```diff
@@ -0,0 +1,123 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.protocol.compatibility;
+
+import com.hazelcast.cache.impl.CacheEventData;
+import com.hazelcast.cache.impl.CacheEventDataImpl;
+import com.hazelcast.cache.impl.CacheEventType;
+import com.hazelcast.client.impl.MemberImpl;
+import com.hazelcast.client.impl.client.DistributedObjectInfo;
+import com.hazelcast.core.Member;
+import com.hazelcast.internal.serialization.impl.HeapData;
+import com.hazelcast.map.impl.SimpleEntryView;
+import com.hazelcast.map.impl.querycache.event.DefaultQueryCacheEventData;
+import com.hazelcast.map.impl.querycache.event.QueryCacheEventData;
+import com.hazelcast.mapreduce.JobPartitionState;
+import com.hazelcast.mapreduce.impl.task.JobPartitionStateImpl;
+import com.hazelcast.nio.Address;
+import com.hazelcast.nio.serialization.Data;
+import com.hazelcast.transaction.impl.xa.SerializableXID;
+
+import javax.transaction.xa.Xid;
+import java.lang.reflect.Array;
+import java.net.UnknownHostException;
+import java.util.AbstractMap;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Map;
+
+public class ReferenceObjects {
+
+    public static boolean isEqual(Object a, Object b) {
+        if (a == b) {
+            return true;
+        }
+        if (a == null || b == null) {
+            return false;
+        }
+        if (a.getClass().isArray() && b.getClass().isArray()) {
+
+            int length = Array.getLength(a);
+            if (length > 0 && !a.getClass().getComponentType().equals(b.getClass().getComponentType())) {
+                return false;
+            }
+            if (Array.getLength(b) != length) {
+                return false;
+            }
+            for (int i = 0; i < length; i++) {
+                if (!isEqual(Array.get(a, i), Array.get(b, i))) {
+                    return false;
+                }
+            }
+            return true;
+        }
+        if (a instanceof List && b instanceof List) {
+            ListIterator e1 = ((List) a).listIterator();
+            ListIterator e2 = ((List) b).listIterator();
+            while (e1.hasNext() && e2.hasNext()) {
+                Object o1 = e1.next();
+                Object o2 = e2.next();
+                if (!isEqual(o1, o2)) {
+                    return false;
+                }
+            }
+            return !(e1.hasNext() || e2.hasNext());
+        }
+        return a.equals(b);
+    }
+
+    public static boolean aBoolean = true;
```

Comment:
I covered all that are used in codecs. Others are defined in protocol but not used apperantly. Since they are not used, I can not add them to these tests. I think they should be added when they are used by a codec.